---------------------
PatchSet 10318 
Date: 2006/05/12 15:57:23
Author: hno
Branch: SQUID_2_5
Tag: (none) 
Log:
Bug #1532: Mime icons are not displayed when viewing ftp sites when
visible_hostname is a short hostname (without domain).

Members: 
	src/internal.c:1.23.2.2->1.23.2.3 

Index: squid/src/internal.c
===================================================================
RCS file: /cvsroot/squid/squid/src/internal.c,v
retrieving revision 1.23.2.2
retrieving revision 1.23.2.3
diff -u -r1.23.2.2 -r1.23.2.3
--- squid/src/internal.c	10 Mar 2006 22:43:37 -0000	1.23.2.2
+++ squid/src/internal.c	12 May 2006 15:57:23 -0000	1.23.2.3
@@ -135,6 +135,9 @@
 {
     LOCAL_ARRAY(char, host, SQUIDHOSTNAMELEN + 1);
     xstrncpy(host, getMyHostname(), SQUIDHOSTNAMELEN);
+    if (Config.appendDomain && !strchr(host, '.'))
+	strncat(host, Config.appendDomain, SQUIDHOSTNAMELEN -
+	    strlen(host) - 1);
     Tolower(host);
     return host;
 }
